### DISCUSSION TITLE
Building Windows binaries (and others eventually)

### DISCUSSION THREAD

**Discussion by axtens** [unoplatform/uno#13632] [Q&A]
The story so far: we hired a programmer (let's call him PRG) on Upwork and asked him to build us a Uno-based IDE for our in-house programming language. We specified that it should be, at least in principle, able to run on Windows, Linux, Mac and Mobile. We set various goals which he met. Now we come to the point where we have bare-bones functionality which we'd like to test on Windows, Linux and Mac.

PRG was using Visual Studio 2022 Professional. We are using Visual Studio Community. Maybe that matters, I don't know. 

We had been struggling to find a way of generating at least Windows EXEs until this morning when I found on unoplatform/uno a command for running MsBuild against the solution file and this generated some EXEs. The command was 
`msbuild "IDE.sln" /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64" /p:AppxBundle=Always /p:UapAppxPackageBuildMode=StoreUpload /p:AppxPackageDir="C:/Code/AppxPackages/" /p:AppxPackageSigningEnabled=false `

I found an EXE at `C:\Users\bugma\Source\Repos\3rdParty\IDE\IDE.Windows\bin\x64\Debug\net7.0-windows10.0.19041.0\win-x64\IDE.Windows.exe`

When I tried to launch the EXE it crashed saying it has an issue with 'Microsoft.UI.Input.dll'. The Event Viewer shows an Error as below: 
```
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" /> 
  <EventID>1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x8000000000000000</Keywords> 
  <TimeCreated SystemTime="2023-09-14T06:47:33.1806705Z" /> 
  <EventRecordID>5442</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="32680" ThreadID="13512" /> 
  <Channel>Application</Channel> 
  <Computer>LAPTOP-H6HBEGA9</Computer> 
  <Security UserID="S-1-5-21-3535587509-2868706409-2911476670-1001" /> 
  </System>
- <EventData>
  <Data Name="AppName">IDE.Windows.exe</Data> 
  <Data Name="AppVersion">1.0.0.0</Data> 
  <Data Name="AppTimeStamp">64e841c5</Data> 
  <Data Name="ModuleName">Microsoft.UI.Input.dll</Data> 
  <Data Name="ModuleVersion">10.0.22624.1038</Data> 
  <Data Name="ModuleTimeStamp">7a840115</Data> 
  <Data Name="ExceptionCode">c000041d</Data> 
  <Data Name="FaultingOffset">0000000000061d14</Data> 
  <Data Name="ProcessId">0x8b24</Data> 
  <Data Name="ProcessCreationTime">0x1d9e6d7547eddcb</Data> 
  <Data Name="AppPath">C:\Users\bugma\Source\Repos\3rdParty\IDE\IDE.Windows\bin\x64\Debug\net7.0-windows10.0.19041.0\win-x64\IDE.Windows.exe</Data> 
  <Data Name="ModulePath">C:\Users\bugma\Source\Repos\3rdParty\IDE\IDE.Windows\bin\x64\Debug\net7.0-windows10.0.19041.0\win-x64\Microsoft.UI.Input.dll</Data> 
  <Data Name="IntegratorReportId">3ff0d035-4a63-4a94-9e3b-92d5ac144011</Data> 
  <Data Name="PackageFullName" /> 
  <Data Name="PackageRelativeAppId" /> 
  </EventData>
  </Event>
```
Where to from here?

**Comment by jeromelaban** [comment]
Thanks for the report. 

Packaging apps for Windows is detailed  [here](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-packaged-apps) and [here](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/single-project-msix?tabs=csharp). You're packaging your application using an msix format, which means that you'll need to install the msix first and run the app from the start menu rather than the exe file itself.

For unpackaged apps, see [this guide](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-unpackaged-apps).
